src/richie/apps/courses/api.py

```diff
@@ -45,3 +54,147 @@ def get_permissions(self):
+
+    - An existing course run is updated only if its "sync_mode" field is set to something else
+      than "manual".
+
+    - The public version of a course run is updated only if its "sync_mode" field is set to
+      "sync_to_public". Otherwise, only the draft version is updated and the related course
+      is marked dirty.
+
+    - A new course run is created only if the "DEFAULT_COURSE_RUN_SYNC_MODE" parameter is set
+      to something else than "manual" in the lms configuration (or the setting
+      "RICHIE_DEFAULT_COURSE_RUN_SYNC_MODE" in the absence of LMS preference). Otherwise, only
+      existing course runs are updated.
+
+    - A new public course run is created only if the "DEFAULT_COURSE_RUN_SYNC_MODE" parameter
+      is set to "sync_to_public" in the lms configuration (or the setting
+      "RICHIE_DEFAULT_COURSE_RUN_SYNC_MODE" in the absence of LMS preference). Otherwise, only
+      the draft course run is created and the related course is marked dirty.
+
+    Parameters
+    ----------
+    request : Type[django.http.request.HttpRequest]
+        The request on the API endpoint, it should contain a payload with course run fields.
+
+    Returns
+    -------
+    Type[rest_framework.response.Response]
+        HttpResponse acknowledging the success or failure of the synchronization operation.
+    """
+    msg = request.body.decode("utf-8")
+
+    # Check if the provided signature is valid against any secret in our list
+    #
+    # We need to do this to support 2 or more versions of our infrastructure at the same time.
+    # It then enables us to do updates and change the secret without incurring downtime.
+    authorization_header = request.headers.get("Authorization")
+    if not authorization_header:
+        return Response("Missing authentication.", status=401)
+
+    signature_is_valid = any(
+        authorization_header
+        == hmac.new(
+            secret.encode("utf-8"), msg=msg.encode("utf-8"), digestmod=hashlib.sha256
+        ).hexdigest()
+        for secret in getattr(settings, "COURSE_RUN_SYNC_SECRETS", [])
+    )
+
+    if not signature_is_valid:
+        return Response("Invalid authentication.", status=403)
+
+    # Select LMS from resource link
+    resource_link = request.data.get("resource_link")
+    if not resource_link:
+        return Response({"resource_link": ["This field is required."]}, status=400)
+
+    lms = LMSHandler.select_lms(resource_link)
+    if lms is None:
+        return Response(
+            {"resource_link": ["No LMS configuration found for this resource link."]},
+            status=400,
+        )
+    sync_mode = lms.default_course_run_sync_mode
+
+    target_course_runs = CourseRun.objects.filter(resource_link=resource_link)
+    draft_course_runs = target_course_runs.filter(draft_course_run__isnull=True)
+
+    serializer = lms.get_course_run_serializer(
+        request.data, partial=bool(draft_course_runs)
+    )
+
+    if serializer.is_valid() is not True:
```

Comment:
`is not True` is the most restrictive... so the least likely to happen. For example, if the call to is_valid() was returning "toto" or None, the test you propose would consider that the form is valid. I feel more confortable to request precisely `True` since anyway this is what it should return.

Reply:
and what about `is False` ?

Reply:
less good because it would still evaluate successfully is it returns None?

Reply:
This is problematic IMO... you can't trust what should return a function and you can't strictly assert the value **and** the type

Reply:
c'est pas faux :smile: